presto-main/src/main/java/com/facebook/presto/execution/SqlStageExecution.java

```diff
@@ -330,7 +330,7 @@ public synchronized RemoteTask scheduleTask(Node node, int partition)
             // The output buffer depends on the task id starting from 0 and being sequential, since each
             // task is assigned a private buffer based on task id.
             TaskId taskId = new TaskId(stateMachine.getStageId(), nextTaskId.getAndIncrement());
-            task = scheduleTask(node, taskId, splits);
+            task = scheduleTask(node, taskId, splits, -1, -1);
```

Comment:
If these values are optional, it'd be better to model them as OptionalInt instead of using magic "-1".

Reply:
@martint In some cases, number of partitions is not known in advance. E.g. Presto can add tasks for table scan as more nodes become available. I'll update to use `OptionalInt`.